components/server/src/workspace/workspace-factory.spec.db.ts

```diff
@@ -0,0 +1,452 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { TypeORM, WorkspaceDB } from "@gitpod/gitpod-db/lib";
+import { resetDB } from "@gitpod/gitpod-db/lib/test/reset-db";
+import {
+    CommitContext,
+    Organization,
+    PrebuiltWorkspaceContext,
+    PrebuiltWorkspaceState,
+    Project,
+    User,
+    WithCommitHistory,
+    WithPrebuild,
+    Workspace,
+    WorkspaceConfig,
+    WorkspaceImageSource,
+} from "@gitpod/gitpod-protocol";
+import { Experiments } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
+import * as chai from "chai";
+import { Container, injectable } from "inversify";
+import "mocha";
+import { OrganizationService } from "../orgs/organization-service";
+import { createTestContainer, withTestCtx } from "../test/service-testing-container-module";
+import { ProjectsService } from "../projects/projects-service";
+import { ConfigProvider } from "./config-provider";
+import { UserService } from "../user/user-service";
+import { SYSTEM_USER } from "../authorization/authorizer";
+import { WorkspaceFactory } from "./workspace-factory";
+import { IncrementalWorkspaceService } from "../prebuilds/incremental-workspace-service";
+import { ImageSourceProvider } from "./image-source-provider";
+import { TraceContext } from "@gitpod/gitpod-protocol/lib/util/tracing";
+
+const expect = chai.expect;
+
+@injectable()
+export class MockIncrementalWorkspaceService extends IncrementalWorkspaceService {
+    public async getCommitHistoryForContext(context: CommitContext, user: User): Promise<WithCommitHistory> {
+        throw new Error("Method not implemented.");
+    }
+}
+
+@injectable()
+export class MockImageSourceProvider extends ImageSourceProvider {
+    public imageSource: WorkspaceImageSource = { baseImageResolved: "gitpod/workspace-full:latest" };
+    public async getImageSource(
+        ctx: TraceContext,
+        user: User,
+        context: CommitContext,
+        config: WorkspaceConfig,
+    ): Promise<WorkspaceImageSource> {
+        return this.imageSource;
+    }
+}
+
+describe("WorkspaceFactory", async () => {
+    let container: Container;
+    let db: WorkspaceDB;
+    // let mockImageSourceProvider: MockImageSourceProvider;
+
+    let owner: User;
+    let member: User;
+    // let stranger: User;
+    let org: Organization;
+    let project: Project;
```

Comment:
Looks like it's fine to delete these two
```suggestion
    let container: Container;
    let db: WorkspaceDB;

    let owner: User;
    let member: User;
    let org: Organization;
    let project: Project;
```